account_invoice_send_method/models/account_invoice.py

```diff
@@ -0,0 +1,15 @@
+# Copyright 2024 Quartile Limited
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class AccountInvoice(models.Model):
+    _inherit = "account.invoice"
+
+    use_company_invoice = fields.Boolean(
+        related="partner_id.use_company_invoice", store=True
```

Comment:
This kind of implementation (`related` between master record and transactions) is generally not advisable as it triggers a mass update when a master record is updated.  Please extend the _onchange_partner_id() method instead.

Reply:
I used create() and write() instead _onchange_partner_id() because using onchange will not work in some scenario.

Reply:
Are you sure about that?

https://github.com/qrtl/odoo/blob/b18fe15/addons/account/models/account_invoice.py#L546

Reply:
Yes. It will triggered but values will not be assigned if we don't add these field in form view. With our current requirement, these fields are not displayed in form view. User will just fill in partner master data. So, instead of extending onchange and adding view unnecessary view definition, I decided to use `create()` and `write()`.